packages/host/app/components/prerendered-card-search.gts

```diff
@@ -127,6 +129,20 @@ export default class PrerenderedCardSearch extends Component<Signature> {
     }
   }
 
+  // we want our query to be reactive in a deepEqual sense vs a strict equal sense
+  @use private query = resource(() => {
+    let query = new TrackedObject<{ value: Query | undefined }>({
+      value: undefined,
+    });
+    (async () => {
+      await Promise.resolve(); // buffer 1 microtask to prevent re-render cycles
+      if (!isEqual(this.args.query, query.value)) {
+        query.value = this.args.query;
+      }
+    })();
+    return query;
+  });
+
```

Comment:
We make a similar guard in PrerenderedSearch as well which has the same vulnerability